### PR TITLE
fix(tests): guard imaginary-part assertions for real tensor types

### LINKED ISSUE #30: tests: add systematic imaginary-part verification for complex type variants
## Summary

Out of 107 test functions, only 6 use `if constexpr` to branch on whether `elem_t` is complex. The remaining 101 tests either skip imaginary-part verification entirely or handle it inconsistently. This means complex type variants (`complex<float>`, `complex<double>`) are not thoroughly validated.

## Current state

Tests with complex-aware assertions (6/107):
- `test_random_inplace`, `test_random_outofplace`
- `test_copy_inplace`, `test_copy_single_element`, `test_copy_large`
- `test_convert_data_integrity`

All other tests that create elements with `make_elem<TenT>(real, imag)` silently discard the imaginary component for complex types when only checking `real_part`.

### Pattern to apply systematically

For every test that stores and retrieves element values, add imaginary-part verification for complex types:

```cpp
auto elem = tci::get_elem(fix.ctx, tensor, coors);
TCICT_ASSERT_CLOSE(real_part<TenT>(elem), expected_real, eps);
if constexpr (is_complex_v<TenT>) {
  TCICT_ASSERT_CLOSE(imag_part<TenT>(elem), expected_imag, eps);
}
```

### Categories of tests to audit

1. **Construction tests** (`fill`, `assign_from_range`, `eye`, `zeros`): verify complex elements are stored correctly
2. **Manipulation tests** (`reshape`, `transpose`, `extract_sub`, `replace_sub`, `concatenate`, `stack`, `for_each`): verify imaginary parts survive transformations
3. **Linear algebra tests** (`scale`, `linear_combine`, `contract`, `diag`): verify complex arithmetic correctness
4. **I/O tests** (`save`/`load`): verify imaginary parts survive serialization round-trip

### Helper trait

Add to `elem_helper.h`:
```cpp
template <typename TenT>
inline constexpr bool is_complex_v = std::is_same_v<tci::elem_t<TenT>, tci::cplx_t<TenT>>;
```

## Scope

- Audit all 107 tests for missing imaginary-part checks
- Add `is_complex_v` helper trait
- Add `if constexpr (is_complex_v<TenT>)` blocks where imaginary parts should be verified
- Use non-zero imaginary values in test inputs where currently only real values are used with complex types

### PR DESCRIPTION
**Pull request by ultimatile**
## Summary

- Test template functions for complex-specific operations broke when instantiated with real `elem_t`: `imag_part<TenT>(x)` assertions against non-zero expectations always failed, one `tci::to_cplx`-based test triggered a type mismatch, and one `Elem(x, y)` constructor was ill-formed for real scalars.
- Add `is_complex_v<TenT>` trait to `elem_helper.h` and wrap every previously-unguarded `imag_part<TenT>` assertion in `if constexpr (is_complex_v<TenT>) { ... }`.
- For `test_imag_extraction` and `test_real_imag_inplace`, assertions are branched (real path verifies spec-mandated zero / identity behavior for `tci::imag` and `tci::real`) rather than entirely skipped — so broken real-tensor implementations are still caught.
- `test_to_cplx_complex_to_complex` is fully body-guarded (for real `TenT` the expression `imag_part<TenT>(cplx_elem)` is a compile-time type mismatch against `elem_t<TenT>`).
- `test_cplx_conj_{in,out}ofplace` keep real-side coverage: since `cplx_conj` on real tensors is a spec no-op / deep copy, `real_part` assertions run for both branches and only `imag_part` assertions are gated.
- Replace `Elem(1.0, 0.0) / elem` with `make_elem<TenT>(1.0, 0.0) / elem` in `test_for_each_inversion`.
- Unify the 6 existing `std::is_same_v<tci::elem_t<TenT>, tci::cplx_t<TenT>>` uses to the new `is_complex_v<TenT>`.

Closes #30.

## Test plan

- [x] cytnx downstream rebuilds cleanly with the modified headers.
- [x] No regression on existing complex tests (baseline: 133 pass / 3 fail / 1 skip, 6293/6293 assertions — identical to pre-change baseline; the 3 failures are pre-existing and unrelated: two `/tmp` file I/O fixture issues and one `tci::order` check).
- [ ] Empirical real-type validation arrives with #31, which wires up real-type registration. At that point the spec-mandated real paths (`tci::imag(real) = 0`, `tci::real(real) = identity`, `tci::cplx_conj(real)` = no-op/deep-copy) will be exercised for the first time.

## Notes

- Adding non-zero imaginary inputs to tests that currently use only real values (the "Use non-zero imaginary values ..." bullet in #30) is deferred to follow-up issues — that work requires per-test design decisions beyond the mechanical guard pattern applied here.

### CODE REVIEW

File: include/tcict/tests/tensor_manipulation.h [hunk 1]
```diff
@@ -185,14 +194,25 @@ void test_real_imag_inplace(tci_test_fixture<TenT> &fix) {
   tci::imag(ctx, tensor, imag_output);
 
   using RealTenT = tci::real_ten_t<TenT>;
+  // Real side: tci::real copies the real parts (== the stored values, since
+  // make_elem drops imag for real TenT; identity for complex TenT).
```
[the remark this answers is not in the document]
Comment (ultimatile) [reply include/tcict/tests/tensor_manipulation.h]: Fixed. The comment incorrectly said 'identity for complex TenT' — the semantics are the opposite: tci::real is a deep copy (identity) for real TenT and extracts real components for complex TenT. Updated the comment accordingly.
